quantumdb-postgresql/src/main/java/io/quantumdb/core/planner/PostgresqlMigrationPlanner.java

```diff
@@ -318,26 +319,19 @@ private Plan createPlan() {
 
 				Set<Step> dependentOnSteps = dependentOnTables.stream()
 						.map(plan::findFirstCopy)
-						.flatMap(o -> o.isPresent() ? Stream.of(o.get()) : Stream.empty())
+						.flatMap(o -> o.map(Stream::of).orElseGet(Stream::empty))
 						.collect(Collectors.toSet());
 
 				Map<Step, Set<Step>> mapping = dependentOnSteps.stream()
 						.collect(Collectors.toMap(Function.identity(), Step::getTransitiveDependencies));
 
 				while (!mapping.isEmpty()) {
-					Optional<Entry<Step, Set<Step>>> crucial = mapping.entrySet().stream()
-							.sorted(Comparator.comparing(entry -> entry.getValue().size() * -1))
-							.findFirst();
+					Optional<Entry<Step, Set<Step>>> crucial = mapping.entrySet().stream().min(Comparator.comparing(entry -> entry.getValue().size() * -1));
 
-					if (crucial.isPresent()) {
-						Entry<Step, Set<Step>> entry = crucial.get();
-						Step step = entry.getKey();
-						dependentOnSteps.removeAll(step.getTransitiveDependencies());
-						mapping.remove(step);
-					}
-					else {
-						break;
-					}
+					Entry<Step, Set<Step>> entry = crucial.get();
+					Step step = entry.getKey();
+					dependentOnSteps.removeAll(step.getTransitiveDependencies());
+					mapping.remove(step);
```

Comment:
If (crucial.isPresent()) can be removed as above you check if mapping is not empty, therefore guaranteeing that a value is present.